app/models/checklist_assignment.rb

```diff
@@ -26,7 +27,19 @@ class ChecklistAssignment < ApplicationRecord
   scope :completed, -> { where.not(completed_at: nil) }
   scope :incomplete, -> { where(completed_at: nil) }
 
+  scope :required, -> { joins(:checklist_template_item).where(checklist_template_items: {required: true}) }
+  scope :optional, -> { joins(:checklist_template_item).where(checklist_template_items: {required: false}) }
+
+  before_create :set_due_date
+  delegate :name, :description, :expected_duration_days, :required, :required?, :optional?, to: :checklist_template_item
+
   def completed?
     completed_at
   end
+
+  private
+
+  def set_due_date
+    self.due_date = Time.now + expected_duration_days.days
```

Comment:
```suggestion
    self.due_date ||= Time.now + expected_duration_days.days
```
In case, someone manually adds them (in the future maybe) then we could skip this OR we could add a `if:` to only run this if the due_date is missing.